File: 1614.cpp
class Solution {
public:
    int maxDepth(string s) {
        int count = 0, maxAns = 0;
        for (char i : s)
        {
            if (i == '(')
            {
                count ++;
                if (count > maxAns) maxAns ++;
            }
            if (i == ')') count--;
        }
        return maxAns;
    }
};

File: 1614.cs
public class Solution {
    public int MaxDepth(string s) {
        int count = 0, maxAns = 0;
        foreach (char i in s){
            if (i == '(') maxAns = Math.Max(maxAns, ++count);
            if (i == ')') count--;
        }
        return maxAns;
    }
}